lib/core/src/lib/datatable/data/object-datatable-adapter.ts

```diff
@@ -54,10 +57,14 @@ export class ObjectDataTableAdapter implements DataTableAdapter {
         return schema;
     }
 
-    constructor(data: any[] = [], schema: DataColumn[] = []) {
+    constructor(data: any[] = [], schema: DataColumn[] = [], sortingMode: SortingMode = 'client') {
         this._rows = [];
         this._columns = [];
 
+        if (!sortingMode) {
```

Comment:
this might be a breaking change if components rely on the client sorting by default... needs to be somehow verified with ACA/ADW probably. cc @MichalKinas

Reply:
@DenysVuika This shouldn't break sorting for any components that don't explicitly set the sortingMode to `'server'`. From my understanding, currently this will only affect the task and process list adapters.

Reply:
This not entirely true since at least search results page in ACA is using the 'server' mode and it might be set through the user preferences also for other lists as far as I can see, @amolodyh-hyland before merging the PR please create linked PR in ACA to see if all tests will pass

Reply:
@MichalKinas Could you point me to which search results you're referring to? I just want to take a look at what adapter it's using.

Reply:
@MichalKinas If you're referring to the document list table then that one uses its own implementation and this update wont affect it.

Reply:
Yes it's about the table in the document list

Reply:
@MichalKinas Should I still run a branch with link-adf since the documents list doesn't use the `ObjectDataTableAdapter`?

Reply:
That's great but this is a PR in HX monorepo and Denys wanted to check https://github.com/Alfresco/alfresco-content-app which is referred to as ACA while adding the comment

Reply:
Ah, I see. Wasn't aware of that repo. I'll get the build ran over there as well then.

Reply:
@MichalKinas I created a PR with a linked commit and everything passed here: https://github.com/Alfresco/alfresco-content-app/pull/4278

However, I'm not 100% sure if that's the correct way to link the ADF library to the ACA PR. Is that the right way to achieve this?

Reply:
The commit message was correct but for some reason ADF was not linked, I pushed new empty commit to your branch and now it seemed to work which you can observe here: https://github.com/Alfresco/alfresco-content-app/actions/runs/12160366230/job/33912497738?pr=4278#step:4:269, lets see if PR will pass now

Reply:
Looks like the build succeeded, minus the "Finalize" step. The finalize step looks like it failed because of the adf-link commit (I'm assuming this is expected?).

Reply:
Yes this is expected, all good on ACA side

Reply:
Looks good to me, when Denys approves you can merge it